src/app/InteractionModelDelegate.h

```diff
@@ -75,6 +76,38 @@ class InteractionModelDelegate
      */
     virtual CHIP_ERROR ReportError(const ReadClient * apReadClient, CHIP_ERROR aError) { return CHIP_ERROR_NOT_IMPLEMENTED; }
 
+    /**
+     * Notification that a Command Send receive Status code embeded in Invoke Command Response
+     * @param[in]  apCommandSender A current command sender which can identify the command sender to the consumer, particularly
+     * during multiple command interactions
+     * @param[in]  aGeneralCode   Status code defined by the standard
+     * @param[in]  aProtocolId    Protocol Id
+     * @param[in]  aProtocolCode  Detailed error information, protocol-specific.
+     * @param[in]  aEndpointId    Endpoint identifier
+     * @param[in]  aClusterId     Cluster identifier
+     * @param[in]  aCommandId     Command identifier
+     * @retval # CHIP_ERROR_NOT_IMPLEMENTED if not implemented
+     */
+    virtual CHIP_ERROR CommandResponseStatus(const CommandSender * apCommandSender, const uint16_t aGeneralCode,
+                                             const uint32_t aProtocolId, const uint16_t aProtocolCode, chip::EndpointId aEndpointId,
+                                             const chip::ClusterId aClusterId, chip::CommandId aCommandId)
```

Comment:
This doesn't really work out if the same sender is used to send two toggle commands, say; you can't tell which one failed.  Or, for cases when it matters more which one failed, if you send two commands that take arguments and have different arguments....

Fundamentally, it seems like we should either have a single CommandSender per command invocation or we need a different way of identifying which exact command failed here.

Reply:
assume there is no protocol error, since we know the status list is exactly same sequence and index with previous sent command list.
per chat, we use that index to notify the upper layer, which command with same command Id is good or not